projects/scannet_offline_eval/build_sparse_voxel_map.py

```diff
@@ -0,0 +1,475 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+    Placeholder code to create 3D map from a scannet scene
+    Currently not functional
+"""
+import argparse
+import dataclasses
+import sys
+import timeit
+from enum import IntEnum, auto
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union
+
+import click
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from evaluation.obj_det import eval_bboxes_and_print
+from torch import Tensor
+from tqdm import tqdm
+
+from home_robot.agent.ovmm_agent.ovmm_perception import OvmmPerception
+from home_robot.core.interfaces import Observations
+from home_robot.datasets.scannet import ScanNetDataset
+from home_robot.mapping.semantic.instance_tracking_modules import Instance
+from home_robot.mapping.voxel import SparseVoxelMap
+from home_robot.perception.constants import RearrangeDETICCategories
+from home_robot.utils.bboxes_3d import (
+    box3d_intersection_from_bounds,
+    box3d_volume_from_bounds,
+)
+from home_robot.utils.config import get_config
+from home_robot.utils.point_cloud_torch import get_bounds
+from home_robot.utils.voxel import VoxelizedPointcloud
+
+
+class SemanticVocab(IntEnum):
+    FULL = auto()
+    SIMPLE = auto()
+    ALL = auto()
+
+
+class SparseVoxelMapAgent:
```

Comment:
This agent should probably move into home_robot/mapping or home_robot/datasets/scannet right?